src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -343,30 +385,29 @@ extern "C" ResultCode GlobalizationNative_GetSortHandle(const char* lpLocaleName
 
     if (U_FAILURE(err))
     {
-        if ((*ppSortHandle)->regular != nullptr)
-            ucol_close((*ppSortHandle)->regular);
-
-        delete (*ppSortHandle);
-        (*ppSortHandle) = nullptr;
+        free(*ppSortHandle);
+        (*ppSortHandle) = NULL;
```

Comment:
I can see that the original code was not doing that, but we are missing pthread_mutex_destroy call here - the CreateSortHandle has created the mutex.